### PR TITLE
created the MVP skeleton for the project

### PR DESCRIPTION
**Pull request by CarlosMiguelCuevas**
DI is missing, and there will be changed when working in specific screens

update git ignore, update gradle plugin for android

add packages for every view, add application class, create base contract

add Rx dependency. format gradle file, saved library version in variables. add MVP files and create packages for every view

create contract per view-presenter in every package, and implemented its methods

### CODE REVIEW

File: app/src/main/java/rraya/nearsoft/com/timesheetsapp/TimeSheetsApp.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package rraya.nearsoft.com.timesheetsapp
+
+import android.app.Application
+
+/**
```
Comment (Mathreyu) [inline app/src/main/java/rraya/nearsoft/com/timesheetsapp/TimeSheetsApp.kt]: Try deleting this from the files. Will do the same :)

File: app/src/main/java/rraya/nearsoft/com/timesheetsapp/data/models/TimesheetItem.kt [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+package rraya.nearsoft.com.timesheetsapp.data.models
 
 data class TimesheetItem(val date: String = "",
                          val other: String = "",
                          val otherHours: Int = 0,
                          val totalHours: Int = 0,
                          val hasHoliday: Boolean = false,
                          val workHours: List<Int>?,
-                         val PTOHours: Int = 0,
+                         val ptoHours: Int = 0,
```
Comment (Mathreyu) [inline app/src/main/java/rraya/nearsoft/com/timesheetsapp/data/models/TimesheetItem.kt]: thanks :D

File: app/src/main/java/rraya/nearsoft/com/timesheetsapp/TimeSheetsApp.kt (hunk 1 above)
Comment (CarlosMiguelCuevas) [reply app/src/main/java/rraya/nearsoft/com/timesheetsapp/TimeSheetsApp.kt]: sorry, didn't see this until now, I'll delete those in the next PR

File: build.gradle [hunk 3]
```diff
@@ -25,3 +25,25 @@ allprojects {
 task clean(type: Delete) {
     delete rootProject.buildDir
 }
+
+ext {
```
Comment (epool) [inline build.gradle]: @CarlosMiguelCuevas you can move this inside of `allProjects` tag like
```groovy
allprojects {
...
    project.ext {
        //SDK and tools
        minSdkVersion = 19
        ...
    }
}
```
with this you can access directly the variable like `$minSdkVersion` or `${minSdkVersion}` if sohuld go inside a string value `"${minSdkVersion}"`

Comment (CarlosMiguelCuevas) [reply build.gradle]: thanks, I'll modify that in the next PR

**Pull request merged**